### DISCUSSION TITLE
How to detect if drawing is inside a svg path and avoid drawing outside

### DISCUSSION THREAD

**Discussion by mehdikarimifa** [fabricjs/fabric.js#8676] [Q&A]
I'm new to fabric.js and trying to implement a simple drawing (coloring) app which loads a simple svg file with 3 circles (e.g two circles inside a bigger circle like a head and eyes). User is able to draw in canvas. If drawing is outside of svg, drawing would be on background, now if drawing is inside the svg, I need to detect the current path edges (like inside the the left eye) and avoid drawings to pass these edges.
What is the best approach?
I appreciate your help.

![image](https://user-images.githubusercontent.com/77890959/217497614-114d25fb-0fd8-42e0-b60c-f952e9c7975d.png)

**Comment by ShaMan123** [comment]
I think clipping/masking is your best option
brushes don't support clip paths yet but soon they will
you can impl yourself meanwhile #8476
